pydicom/pixel_data_handlers/rle_handler.py

```diff
@@ -0,0 +1,219 @@
+        format_str = 'int{}'.format(dicom_dataset.BitsAllocated)
+    else:
+        format_str = 'bad_pixel_representation'
+
+    try:
+        numpy_format = numpy.dtype(format_str)
+    except TypeError:
+        msg = ("Data type not understood by NumPy: "
+               "format='{}', PixelRepresentation={}, "
+               "BitsAllocated={}".format(
+                   format_str,
+                   dicom_dataset.PixelRepresentation,
+                   dicom_dataset.BitsAllocated))
+        raise TypeError(msg)
+
+    UncompressedPixelData = bytearray()
+
+    if ('NumberOfFrames' in dicom_dataset and
+            dicom_dataset.NumberOfFrames > 1):
+
+        CompressedPixelDataSeq = pydicom.encaps.decode_data_sequence(
+            dicom_dataset.PixelData)
+
+        for frame in CompressedPixelDataSeq:
+            decompressed_frame = _rle_decode_frame(frame,
+                                                   rows=dicom_dataset.Rows,
+                                                   columns=dicom_dataset.Columns,  # noqa
+                                                   samples_per_pixel=dicom_dataset.SamplesPerPixel,  # noqa
+                                                   bits_allocated=dicom_dataset.BitsAllocated)  # noqa
+
+            UncompressedPixelData.extend(decompressed_frame)
+
+    else:
+
+        CompressedPixelData = pydicom.encaps.defragment_data(
+            dicom_dataset.PixelData)
+
+        decompressed_frame = _rle_decode_frame(CompressedPixelData,
+                                               rows=dicom_dataset.Rows,
+                                               columns=dicom_dataset.Columns,
+                                               samples_per_pixel=dicom_dataset.SamplesPerPixel,  # noqa
+                                               bits_allocated=dicom_dataset.BitsAllocated)  # noqa
+
+        UncompressedPixelData.extend(decompressed_frame)
+
+    pixel_array = numpy.frombuffer(UncompressedPixelData, numpy_format)
+
+    return pixel_array
+
+
+def _rle_decode_frame(d, rows, columns, samples_per_pixel, bits_allocated):
+    """Decodes a single frame of RLE encoded data.
+    Reads the plane information at the beginning of the data.
+    If more than pixel size > 1 byte appropriately interleaves the data from
+    the high and low planes. Data is always stored big endian. Output always
+    little endian
+
+    Parameters
+    ----------
+    d: bytes
+        The RLE frame data
+    Rows: int
+        The number of output rows
+    Columns: int
+        The number of output columns
+    SamplesPerPixel: int
+        Number of samples per pixel (e.g. 3 for RGB data).
+    BitsAllocated: int
+        Number of bits per sample - must be a multiple of 8
+
+    Returns
+    -------
+    bytearray
+        The decompressed data
+    """
+
+    rle_start = 0
+    rle_len = len(d)
+
+    number_of_planes = unpack(b'<L', d[rle_start: rle_start + 4])[0]
+
```

Comment:
keeing in the rle_start (even thoguh it is 0 currently) as in future may pass in PixelData not cut up into frames (to save reading it all when you just want the first frame).